src/entities/DistributionCampaignBase.ts

```diff
@@ -0,0 +1,131 @@
+import { BigintIsh, ChainId, SECONDS_IN_YEAR } from '../constants'
+import JSBI from 'jsbi'
+import { parseBigintIsh } from '../utils'
+import { CurrencyAmount, Fraction, Percent, TokenAmount } from './fractions'
+import { PricedTokenAmount } from './fractions/priced-token-amount'
+import invariant from 'tiny-invariant'
+import { Token } from './token'
+import { utils } from 'ethers'
+
+
+const MINIMUM_STAKED_AMOUNT_NATIVE_CURRENCY: { [chainId in ChainId]: CurrencyAmount } = {
+  [ChainId.RINKEBY]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('0.05', Token.getNative(ChainId.RINKEBY).decimals).toString(),
+    ChainId.RINKEBY
+  ),
+  [ChainId.MAINNET]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('0.1', Token.getNative(ChainId.MAINNET).decimals).toString(),
+    ChainId.MAINNET
+  ),
+  [ChainId.XDAI]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('1000', Token.getNative(ChainId.XDAI).decimals).toString(),
+    ChainId.XDAI
+  ),
+  [ChainId.ARBITRUM_ONE]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('0.1', Token.getNative(ChainId.ARBITRUM_ONE).decimals).toString(),
+    ChainId.ARBITRUM_ONE
+  ),
+  [ChainId.ARBITRUM_RINKEBY]: CurrencyAmount.nativeCurrency(
+    utils.parseUnits('0.05', Token.getNative(ChainId.ARBITRUM_RINKEBY).decimals).toString(),
+    ChainId.ARBITRUM_RINKEBY
+  )
+}
+
+
+
+export class DistributionCampaignBase {
+  public readonly chainId: ChainId
+  public readonly address?: string
+  public readonly startsAt: BigintIsh
+  public readonly endsAt: BigintIsh
+  public readonly rewards: PricedTokenAmount[]
+  public readonly staked: PricedTokenAmount
+  public readonly duration: BigintIsh
+  public readonly locked: boolean
+  public readonly stakingCap: TokenAmount
+
+  constructor(
+    startsAt: BigintIsh,
+    endsAt: BigintIsh,
+    rewards: PricedTokenAmount[],
+    staked: PricedTokenAmount,
+    locked: boolean,
+    stakingCap: TokenAmount,
+    address?: string
+  ) {
+    invariant(JSBI.lessThan(parseBigintIsh(startsAt), parseBigintIsh(endsAt)), 'INCONSISTENT_DATES')
+    for (const reward of rewards) {
+      invariant(staked.token.chainId === reward.token.chainId, 'CHAIN_ID')
+    }
+    this.chainId = staked.token.chainId
+    this.startsAt = startsAt
+    this.endsAt = endsAt
+    this.rewards = rewards
+    this.staked = staked
+    this.duration = JSBI.subtract(parseBigintIsh(endsAt), parseBigintIsh(startsAt))
+    this.locked = locked
+    this.stakingCap = stakingCap
+    this.address = address
+  }
+
+  public get remainingDuration(): JSBI {
+    const now = JSBI.BigInt(Math.floor(Date.now() / 1000))
```

Comment:
just curious here: do we use JSBI for cross-browser compatibility?
Because `console.log(Date.now())` eq `1639736032449 -> Number`

Reply:
Ideally, we should not use `BigNumber` libraries such as BigNumber.js, and here JSBI, but it's a kind of convention to use. I would not worry about it.